polaris/polaris/management/commands/watch_transactions.py

```diff
@@ -200,18 +200,22 @@ def update_transaction(
 
     @staticmethod
     def _check_payment_op(
-        operation: Operation, want_asset: Asset, want_amount: Decimal
+        operation: Operation, want_asset: Asset, want_amount: Optional[Decimal]
     ) -> bool:
         # TODO: Add test cases!
         issuer = operation.asset.issuer
         code = operation.asset.code
-        print(want_asset, want_amount)
-        print(operation.destination, want_asset.distribution_account)
-        print(str(issuer), want_asset.issuer)
+
+        # SEP-6 doesn't require 'amount' parameter when creating transaction
+        # so we can't use it to match the transaction. Only use it if present.
+        amounts_match = True
+        if want_amount:
+            amounts_match = want_amount == Decimal(operation.amount)
```

Comment:
Yea that makes sense. This gets into a whole area of tough situations. 

If we remove the check and someone withdraws _way more_ than they specified in the interactive flow, it would be up to the anchor to notice the difference in amounts and raise an error. If they didn't catch it and just sent the amount the user specified in the interactive flow, the user just lost a chunk of their money right?

I guess this is what currently happens when they send an amount that doesn't match, but Polaris should probably handle this case better.

I'll file an issue to explore the options. Maybe we provide more flexibility in the integration to allow anchors to reject withdraws and send them back?

Reply:
https://github.com/stellar/django-polaris/issues/196